clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -195,6 +197,7 @@ public void add(TopicPartition topicPartition, PartitionData data) {
         }
 
         public FetchRequestData build() {
+            fetchInProgress = true;
```

Comment:
Ideally flipping this flag would be managed via try/finally blocks

Reply:
This code has been removed.